File: frontend/src/types/moodBoard.ts
export interface MoodBoard {
  id: number
  name: string
  description: string
  category: string
  itemCount: number
  coverImage: string | null
  tags: string[]
  isPublic: boolean
  createdAt: string
}

export interface MoodBoardItem {
  id: number
  boardId: number
  type: string
  title: string
  imageUrl: string | null
  sourceUrl: string | null
  note: string | null
  color: string | null
  position: number
}

export interface CreateMoodBoardRequest {
  name: string
  description: string
  category: string
  tags: string[]
}

export interface MoodBoardSummary {
  totalBoards: number
  totalItems: number
  topCategory: string
  recentBoard: string
  avgItemsPerBoard: number
}

File: backend/onGo-domain/src/main/kotlin/com/ongo/domain/moodboard/MoodBoard.kt
package com.ongo.domain.moodboard

import java.time.LocalDateTime

data class MoodBoard(
    val id: Long = 0,
    val workspaceId: Long,
    val name: String,
    val description: String? = null,
    val category: String? = null,
    val itemCount: Int = 0,
    val coverImage: String? = null,
    val tags: List<String> = emptyList(),
    val isPublic: Boolean = false,
    val createdAt: LocalDateTime = LocalDateTime.now(),
)
